Store parameters names before full method loading

jeannielynnmoulton opened the pull request:
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

Commit:
Pull out parameter naming into a new function.

This code will be reused earlier too, in
java_bytecode_convert_method_lazy

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c605e5d752f5691d1d35bb4f771e06ec7d75cf9f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/129892909

Commit:
Assign names to parameters in java_bytecode_convert_method_lazy

We want to have the parameter names in the method symbol at this stage,
then add the parameter symbols during conversion.

Commit:
Unit tests create_parameter_names

Commit:
Unit test create_parameter_symbols

smowton commented on jbmc/src/java_bytecode/java_bytecode_convert_method.h:
```diff
@@ -58,4 +59,22 @@ void java_bytecode_convert_method_lazy(
+/// i.e. an integer
+/// \param: variables, a table of local variables
+/// \param: symbol_table, the symbol table
+void construct_parameter_symbols(
```
construct -> create, to avoid ambiguity with class construction

smowton commented on jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:
```diff
@@ -405,13 +405,12 @@ void java_bytecode_convert_method_lazy(
     java_method_parameter_slots(method_type);
```
Commit message: "Only load parameter symbols" -> "Only create parameter symbols"?

smowton commented on jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:
```diff
@@ -397,6 +397,21 @@ void java_bytecode_convert_method_lazy(
+    method_identifier,
+    parameters,
+    slots_for_parameters,
+    dummy_local_variable_table);
```
Use a non-owning pointer (i.e. `expanding_vectort<std::vector<java_bytecode_convert_methodt::variablet>> *`) rather than a dummy. Or, factor the method a little more (`create_parameter_local_variables` for example).

jeannielynnmoulton replied:
Turns out to be unnecessary at the end, so removed.

smowton commented on jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:
```diff
@@ -397,6 +397,21 @@ void java_bytecode_convert_method_lazy(
+  java_method_typet::parameterst &parameters = method_type.parameters();
+  java_bytecode_convert_methodt::method_offsett slots_for_parameters =
+    java_method_parameter_slots(method_type);
+  expanding_vectort<std::vector<java_bytecode_convert_methodt::variablet>>
```
Add a typedef for this to `java_bytecode_convert_method_classt` (I think)

smowton reviewed (approved):
Some tweaks, nothing blocking

Commit:
Remove dummy variable in construct_parameter_names

This is unnecessary, and simply and artifact of my refactoring process.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7b8ca793b84961b10db087b935e706e16473f480).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/130185400
